src/index.ts

```diff
@@ -31,9 +29,12 @@ import type { GraphiQLOptions, GraphiQLData } from './renderGraphiQL';
 import { parseBody } from './parseBody';
 import { renderGraphiQL } from './renderGraphiQL';
 
-type $Request = IncomingMessage;
-type $Response = ServerResponse & {| json?: (data: mixed) => void |};
+type Request = IncomingMessage;
+type Response = ServerResponse & { json?: (data: unknown) => void };
 type MaybePromise<T> = Promise<T> | T;
+type Result = Pick<ExecutionResult, 'data' | 'extensions'> & {
+  errors?: ReadonlyArray<any>;
+};
```

Comment:
Why you need a custom result?

Reply:
`errors` in `ExecutionResult` is typed as `ReadonlyArray<GraphQLError>` but since we allow the errors in whichever way, the actual type needs to be `ReadonlyArray<any>`

Reply:
aaa now I see why you need it because `errors` both `GraphQLError` and JSON object.